utils/src/main/java/software/amazon/awssdk/utils/cache/CachedSupplier.java

```diff
@@ -253,12 +361,32 @@ public Builder<T> prefetchStrategy(PrefetchStrategy prefetchStrategy) {
             return this;
         }
 
+        /**
+         * Configure the way the cache should behave when a stale value is retrieved or when retrieving a value fails while the
+         * cache is stale.
+         *
+         * By default, this uses {@link StaleValueBehavior#STRICT}.
+         */
+        public Builder<T> staleValueBehavior(StaleValueBehavior staleValueBehavior) {
+            this.staleValueBehavior = staleValueBehavior;
+            return this;
+        }
+
+        /**
+         * Configure the clock used for this cached supplier. Configurable for testing.
+         */
+        @SdkTestInternalApi
+        public Builder<T> clock(Clock clock) {
```

Comment:
This is passed in from the InstanceProfileCredentialProvider, so it needs to be public. This is a protected class, though, so it's not something customers will see. Otherwise I wouldn't have done it this way.